### PR TITLE
Fix #2143: Allow environment variables to be substituted in OpenMetadata config

### LINKED ISSUE #2143: Allow environment variables to be substituted in OpenMetadata config
More details can be found in this Issue https://github.com/open-metadata/openmetadata-helm-charts/issues/8

### PR DESCRIPTION
**Pull request by harshach**
see #2143 

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: conf/openmetadata.yaml [hunk 1]
```diff
@@ -105,10 +105,10 @@ database:
   # the name of the JDBC driver, mysql in our case
   driverClass: com.mysql.cj.jdbc.Driver
   # the username and password
-  user: openmetadata_user
-  password: openmetadata_password
+  user: ${MYSQL_USER:- openmetadata_user}
+  password: ${MYSQL_USER_PASSWORD:- openmetadata_password}
   # the JDBC URL; the database is called openmetadata_db
-  url: jdbc:mysql://localhost/openmetadata_db?allowPublicKeyRetrieval=true&useSSL=false&serverTimezone=UTC
+  url: jdbc:mysql://${MYSQL_HOST:- localhost}/${MYSQL_DATABASE:- openmetadata_db}?allowPublicKeyRetrieval=true&useSSL=false&serverTimezone=UTC
```
Comment (akash-jain-10) [inline conf/openmetadata.yaml:111]: should we keep `mysql` as default value for MYSQL_HOST ?

Comment (harshach) [reply conf/openmetadata.yaml:111]: localhost will not work for dev

Comment (amiorin) [inline conf/openmetadata.yaml:111]: I would allow to override the MySQL port too, WDYT?

Comment (harshach) [reply conf/openmetadata.yaml]: done.

**Review by akash-jain-10 — APPROVED** [review]

**Pull request merged**